pkg/sql/schema_changer.go

```diff
@@ -1618,18 +1628,16 @@ func (r schemaChangeResumer) Resume(
 				// If the context was canceled, the job registry will retry the job.
 				// We check for this case so that we can just return the error without
 				// wrapping it in a retry error.
-				return err
-			case !isPermanentSchemaChangeError(err):
+				return scErr
+			case !isPermanentSchemaChangeError(scErr):
 				// Check if the error is on a whitelist of errors we should retry on,
-				// including the schema change not having the first mutation in line,
-				// and have the job registry retry.
-				return jobs.NewRetryJobError(err.Error())
+				// including the schema change not having the first mutation in line.
 			default:
 				// All other errors lead to a failed job.
-				return err
+				return scErr
 			}
 		}
-		return nil
+		return jobs.NewRetryJobError(scErr.Error())
```

Comment:
If you're here it probably means that your context was canceled. It's reasonably like that `scErr` is nil here which means this will panic.

Reply:
well if scErr was nil, we would return inside the body of the loop